### PR TITLE
Add initial version of development guidelines

### PR DESCRIPTION
**Pull request by arnaucube**

### CODE REVIEW

File: src/about-us/development-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+## Intro
+- The 'motto'is: **keep it simple**
+    - If we extend too much the explainations or the 'rules', will be harder to read, and by consequence harder to be followed
+    - This kind of documents can end up being too much 'opinionated', we try to write down the *'minimum common base layer'*, and then if a specific team want to add some extra concrete step is free to do it so.
+
+## Tasks management
+- Description of current status:
+    - Each project uses different tool
+    - Concrete tasks are defined in `Github issues`
+    - General tasks are defined in `Linear` + `Jira` + `Notion` *(document: 'Monday commitments' (priorities of the week & forecast for month))*
```
Comment (ed255) [inline src/about-us/development-guidelines.md:15]:
```suggestion
    - General tasks are defined in
        - `Linear`: *Will be replaced by Jira eventually*
        - `Jira` 
        - `Notion`: *document: 'Monday commitments' (priorities of the week & forecast for month)*
```

File: src/about-us/development-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - TODO: Define code 'owner'/'default reviewer' concept
+    - TODO: Agree on who is knowledgable of each repository package, and write it down --> 'code owner' (enforced by Github)
+- When creating a PR the PR creator needs to `require` at least 1 PR reviewer
+    - Take in mind that the PR reviewer may be receiving lots of emails of notifications of Github, so the PR creator should decide if it's needed to send a private message to that person to notify of the PR
+
+#### Automated Code Review
+- PRs can be merged once:
+    - the tests pass
+    - the linters pass (if activated)
+- It's responsability of the repo creator/owner to configure the Github Actions with the automated Tests & Linters for each commit&PR
```
Comment (ed255) [inline src/about-us/development-guidelines.md:68]:
```suggestion
- It's responsability of the repo creator/owner to configure the Github Actions with the automated Tests & Linters for each commit & PR
```

**Review by ed255 — APPROVED** [review]

**Comment by ed255** [comment]
I've added some minor edit suggestions, but otherwise LGTM

**Review by jordipainan — APPROVED** [review]

**Review by NateWilliams2 — APPROVED** [review]
I've added some linguistic edits, otherwise looks great to me.

File: src/about-us/development-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+> current page is what we are using while the shared doc with Aragon gets done.*
+
+## Intro
+- The 'motto' is: **keep it simple**
+    - If we over-extend the explanations or the 'rules', these guidelines will be harder to read and therefore harder to follow.
+    - Development guidelines can easily fall into the trap of being very opinionated and subjective. To avoid this, we aim to specify a *'minimum common base layer'* for our development. If a specific team wants to add some extra concrete steps particular to their area of expertise, they are free to do so. 
+
+## Tasks management
+- Description of current status:
+    - Each project uses different tool
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:13]: Why this?

File: src/about-us/development-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    - Concrete tasks are defined in `Github issues`
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:14]: GitHub issued only for errors reported by the community, I would say.
Otherwise we use Linear/Jira

File: src/about-us/development-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    - General tasks are defined in
+        - `Linear`: *Will be replaced by Jira eventually*
+        - `Jira` 
+        - `Notion`: *document: 'Monday commitments' (priorities of the week & forecast for month)*
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:18]: Notion we don't use it to define tasks. 
- We use Notion and/or HackMd for internal discussions... and then we reference it from the relevant Linear tasks, or even from Discord. 
- For public discussions we use GitHub discussions on the protocol repo

File: src/about-us/development-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    
+## Docs/Specs strategy
+- Code should be well documented (following the specific language guidelines)
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:21]: Maybe we can elaborate on the minimum content that the project's Readme file should have, as well

File: src/about-us/development-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+- New features should be documented before being implemented
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:22]: I think **protocol definitions** need to be documented.

**Features**... I think they need to be defined on Linear or even on Notion, so the product teams can build them. But not "documented" like the protocol.

This is the role of the product manager, which we may need an equivalent for... in the protocol team.

File: src/about-us/development-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    - Reasons:
+        - Avoid inefficiencies due incompatibilities on the client & server sides
+        - Make it easier for the reviewer to verify that the implementation follows the spec
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:25]: The current methodology is that tasks should always include a "definition of done".

This way, the sprint manager can tell whether something has been achieved or not

File: src/about-us/development-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+### Layers of documentation
```
Comment (brickpop) [inline src/about-us/development-guidelines.md:27]: Maybe this feels more like a general topic that could appear before?

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (p4u) [inline src/about-us/development-guidelines.md:22]: I'd add a reference to discuss.vocdoni.io for discussing new features and so on

**Review by p4u — APPROVED** [review]

File: src/about-us/development-guidelines.md (hunk 3 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:13]: Because is the current status. In this way we can already put the document in the docs, and from here we can iterate if somebody wants to do a new PR adding a proposal.

File: src/about-us/development-guidelines.md (hunk 4 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:14]: I think that this is also a point that needs to be defined. Currently there are different tools and each person/project uses a different one. I would propose to keep it in this way for this PR, and once is defined which tool to use we update the document.

File: src/about-us/development-guidelines.md (hunk 5 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:18]: Currently (included this last Monday) Notion is used for the document of 'Monday commitments', that's why we putted the description saying `Notion: *document: 'Monday commitments' (priorities of the week & forecast for month)*`.
I think that the line reflects the current status & usage in this case.

File: src/about-us/development-guidelines.md (hunk 6 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:21]: We mentioned this on the meeting of this doc, but it seemed to us that it will depend on the kind of repo and also on the language being used, so for the moment we decided to not include it inside this guidelines doc. But if there is a proposal we can work on it.

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:22]: Cool, added

File: src/about-us/development-guidelines.md (hunk 9 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:29]: Currently this topic is under the parent-topic `Docs/Specs strategy`, and after the topics `Intro` and `Tasks management`:
Where you would propose to move the `Layers of documentation` topic?

File: src/about-us/development-guidelines.md (hunk 8 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:25]: Could you add a link to the current methodology definition, so we can be referenced from this document?

File: src/about-us/development-guidelines.md (hunk 3 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:13]: sorry I selected 4 lines. I was referring to `Each project uses different tool`?
By now all projects use Linear, right?

File: src/about-us/development-guidelines.md (hunk 4 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:14]: Well, we defined this a while ago, that's why I'm mentioning this :)
The idea was to use GitHub for its community leverage. So externally reported issues make sense to be received on GitHub. 

But internal tasks, sprint planning, etc all of this happens on Linear (and soon potentially on Jira)

File: src/about-us/development-guidelines.md (hunk 5 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:18]: Ok, maybe it is more straightforward to read if we refer to what type of content is handled where. This Notion document is very specific and I think that something like this could be more helpful:
- `Linear/Jira`: Tasking
- `Notion`: Internal discussion
- `GitHub Discussions`: Public discussion

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:22]: Ok, maybe there's a language mismatch then :)
- "Feature" is often used to talk about something that the product users can do (like product features)
- Tech capabilities may be denominated differently, because it created confusion (at least to me)

File: src/about-us/development-guidelines.md (hunk 8 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:25]: These documents are internal. PM.

File: src/about-us/development-guidelines.md (hunk 9 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:29]: Maybe before `Task Management`?

Comment (arnaucube) [reply src/about-us/development-guidelines.md:29]: Meaning to be outside the `Docs/Specs strategy`?

File: src/about-us/development-guidelines.md (hunk 8 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:25]: Could you add a link to it, so it can be referenced from this document?

File: src/about-us/development-guidelines.md (hunk 5 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:18]: But this point is specifically for: `General tasks are defined in` (as it appears in the line15). It does not mention anything about public/internal discussions.
And being a point on the topic of where are defined the `General tasks`, it mentions the current 3 places where `General tasks` are defined:
- `Linear`: *Will be replaced by Jira eventually*
- `Jira` 
- `Notion`: *document: 'Monday commitments' (where the tasks for the priorities of the week are written down & forecast for month)*

Do you think that this is not the current way of how the `General tasks` are defined?

File: src/about-us/development-guidelines.md (hunk 4 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:14]: > Well, we defined this a while ago, that's why I'm mentioning this :)
Cool, could you add the link of this definition? In this way we can link to the definition that you mention in the guidelines.

File: src/about-us/development-guidelines.md (hunk 3 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:13]: Currently some are using Linear (example: zk-census-proofs), and others Jira (example: devops)

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:22]: Cool, even with both understandings of 'feature', the line 22 can still be true:
- the 'product users features' are documented in Linear/Jira/Notion
- the 'protocol features / tech capabilities' are documented in docs.vocdoni.io

But still the line 22 sentence holds true, right? In both cases the 'feature' should be documented before starting to code it.

File: src/about-us/development-guidelines.md (hunk 3 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:13]: Instead of `Each project uses different tool` I would say something like `Product teams are currently using Jira` and `The Protocol team is still using Linear`

File: src/about-us/development-guidelines.md (hunk 4 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:14]: Verbal agreement :)
I would reword my last comment and explain this same idea

File: src/about-us/development-guidelines.md (hunk 3 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md]: Cool, added a commit to update the line with your suggestion.

File: src/about-us/development-guidelines.md (hunk 5 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:20]: It depends on how you put the question. If you talk about `What's the general place to add tasks?` then the answer would be `It depends on the scope`.

Internal tasks should be created by product leads or protocol leads. If you see a task that needs to be there, you should probably create a Linea/Jira task on the backlog and tell the lead to sort it out.

But for public contributions, issue creation can only happen on GitHub... so there is no general answer to this, and that's why I think we need to explain what to do, depending on the situation.

Comment (arnaucube) [reply src/about-us/development-guidelines.md:20]: Agree. I would propose to keep it as it is now, as it's intended to cover the guidelines for internal devs, and once this PR is merged, we can do a 2nd PR which adds the 'External dev' guidelines part (that can be distributed across the existing sections of the current document.

File: src/about-us/development-guidelines.md (hunk 6 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:23]: I've checked the current pinned repos from github.com/vocdoni, and not all of them follow this sections (the first 3 ones from the list). Also, depending on the repo there is different needs of content in the readme.
I would propose to keep this PR as it is, and encourage you to propose to do a definition (in a meeting or in a new PR) of the readme file template that should be used in all the repos

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:24]: Even if it does, I would be a bit more specific.
- **Product features** need to be well described before being implemented.
   - A description of **why** this change is being made, is expected.
   - A clear "definition of done" is expected.
- **Protocol changes** need to be documented on a docs PR, following suit of a discussion
   - Use the discussions on the `protocol` repo when the topic is relevant to the community
   - Use a Notion page or a HackMD if the topic is relevant to the team only

File: src/about-us/development-guidelines.md (hunk 8 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:27]: See the comment above, with a proposed text: 
https://github.com/vocdoni/docs/pull/43#discussion_r673830102

File: src/about-us/development-guidelines.md (hunk 9 above)
Comment (brickpop) [reply src/about-us/development-guidelines.md:31]: It's an option to me. Going from abstract to more specific. This feels like a gentle onboarding, instead of jumping into specifics and then get back to abstract.

But I'm fine if you like it like this

File: src/about-us/development-guidelines.md (hunk 7 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:24]: Great, added the suggestion in the PR.

File: src/about-us/development-guidelines.md (hunk 8 above)
Comment (arnaucube) [reply src/about-us/development-guidelines.md:27]: https://github.com/vocdoni/docs/pull/43#discussion_r673836007

**Review by brickpop — APPROVED** [review]
